content/hardware/08.edu/solution-and-kits/alvik/tutorials/setting-alvik-arduino-ide/setting-alvik-arduino-ide.md

```diff
@@ -0,0 +1,184 @@
+
+- **Arduino_AlvikCarrier**: This library is designed for the STM board on the device and it is useful in situations where more fine control is required over commands. It allows more complex developments, especially when deeper integration with the hardware is needed. More information and the download link can be found [here](https://www.arduino.cc/reference/en/libraries/arduino_alvikcarrier/).
+
+### Print Firmware Version
+
+A simple but useful program if you are new to the Alvik is to understand how to get information from the onboard STM board. In this case, we are creating a simple sketch that prints the firmware version using the `get_version()` function.
+
+```c++
+#include "Arduino_Alvik.h"
+```
+
+Including the `Arduino_Alvik` library is essensial. The library contains all the needed predefined functions and classes that simplify the process of controlling the robot's hardware, such as motors and sensors. Without this include statement, the compiler wouldn't recognize the `Arduino_Alvik` class or its associated methods.
+
+```c++
+Arduino_Alvik alvik;
+```
+
+When using the Alvik livrary we declare an object of the `Arduino_Alvik` class named `alvik` in this case. All interactions with the robot will go through this `alvik` object. As with our `alvik.drive()` command.
+
+```c++
+void setup() {
+  alvik.begin();
+  Serial.begin(115200);
+}
+```
+
+In the `setup()` function, the `alvik.begin()` method initializes the Alvik robot. This will be a necessity on all sketches for the Alvik.
+
+```c++
+void loop() {
+
+  uint8_t u,m,l;
+
+  alvik.get_version(u,m,l); // Gets the firmware version
+  Serial.printf("%d.%d.%d\n", u, m, l);
+  alvik.drive(10, 45);
+  delay(1000); // Waits a second
+  alvik.drive(10, -45);
+  delay(1000); // Waits a second
+
+}
+```
+
+This code continuously retrieves and prints the firmware version thanks to the ```get_version()``` every second while making the Alvik robot's wheels rotate back and forth using the ```drive()``` command.
+
+
+**Complete code:**
+
+```c++
+#include "Arduino_Alvik.h"
+
+Arduino_Alvik alvik;
+
+void setup() {
+  alvik.begin();
+  Serial.begin(115200);
+}
+
+void loop() {
+
+  uint8_t u,m,l;
+
+  alvik.get_version(u,m,l); // Gets the firmware version
+  Serial.printf("%d.%d.%d\n", u, m, l);
+  alvik.drive(10, 45);
+  delay(1000); // Waits a second
+  alvik.drive(10, -45);
+  delay(1000); // Waits a second
+
+}
+```
+
+### Obstacle Avoider Example In C++
+
+A more complex example can be found in the [getting started](https://docs.arduino.cc/tutorials/alvik/getting-started/) guide for the Alvik.
+Due to the functions having same structure and names on both MicroPython and C++ we can easily port the ```obstacle-avoider``` to C++.
+
+Keeping in mind the initialization of the Alvik from the previous example, we can build the example on the Arduino IDE:
+
+```c++
+#include "Arduino_Alvik.h"
+#include <Arduino.h>
```

Comment:
```suggestion
```
You don't need to add `#include` directives for `Arduino.h` to `.ino` sketch files. The Arduino sketch build system automatically adds that `#include` directive:

https://arduino.github.io/arduino-cli/latest/sketch-build-process/#pre-processing

> - If not already present, `#include <Arduino.h>` is added to the sketch

Reply:
@per1234 Good point! Thanks!